packages/shared_preferences/shared_preferences_ios/lib/SharedPreferencesIos.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'messages.g.dart';
+
+typedef _Setter = Future<bool> Function(String key, Object value);
+
+/// iOS implementation of shared_preferences.
+class SharedPreferencesIos extends SharedPreferencesStorePlatform {
+  final SharedPreferencesApi _api = SharedPreferencesApi();
+  late final Map<String, _Setter> _setters = <String, _Setter>{
+    'Bool': (String key, Object value) {
+      return _api.setBool(key, value as bool);
+    },
+    'Double': (String key, Object value) {
+      return _api.setDouble(key, value as double);
+    },
+    'Int': (String key, Object value) {
+      return _api.setInt(key, value as int);
+    },
+    'String': (String key, Object value) {
+      return _api.setString(key, value as String);
+    },
+    'StringList': (String key, Object value) {
+      return _api.setStringList(key, value as List<String?>);
+    },
```

Comment:
Unfortunately the interface has strings built into it, so I used them as the source of truth.